experimental/AlgebraicShifting/src/AlgebraicShifting.jl

```diff
@@ -0,0 +1,119 @@
+include("UniformHypergraph.jl")
+include("PartialShift.jl")
+include("PartialShiftGraph.jl")
+
+export UniformHypergraph
+
+export compound_matrix
+export contracted_partial_shift_graph
+export exterior_shift
+export face_size
+export generic_unipotent_matrix
+export partial_shift_graph
+export partial_shift_graph_vertices
+export rothe_matrix
+export uniform_hypergraph
+
+function independent_columns(A::MatElem)
+  col_indices = Int[]
+  row_index = 1
+  row_bound = nrows(A)
+  for col_index in 1:ncols(A)
+    if !is_zero(A[row_index, col_index])
+      push!(col_indices, col_index)
+      row_index += 1
+    end
+    row_index > row_bound && break
+  end
+  return col_indices
+end
+
+
+"""
+    symmetric_shift(F::Field, K::SimplicialComplex)
+
+Returns the symmetric shift of `K`
+
+see survey on algebraic shifting Gil Kalai (properly cite at some point)
+"""
+function symmetric_shift(F::Field, K::SimplicialComplex;
+                         change_of_basis::T = nothing) where T <: Union{Nothing, MatElem}
+
+  n = n_vertices(K)
+  is_generic = false
+  if isnothing(change_of_basis)
+    is_generic = true
+    # generic change of basis
+    Fy, y = polynomial_ring(F, :y => (1:n, 1:n))
```

Comment:
Rings created internally should have caching disabled. Please also in all the other places (user facing examples e.g. in docstrings don't need it)
```suggestion
    Fy, y = polynomial_ring(F, :y => (1:n, 1:n); cached=false)
```